src/osvr/Server/ServerImpl.cpp

```diff
@@ -101,9 +100,30 @@ namespace server {
             common::elements::StringElement(util::makeString(display_json));
         // Deal with updated device descriptors.
         m_conn->registerDescriptorHandler([&] { m_handleDeviceDescriptors(); });
+
+        // Set up handlers to enter/exit idle sleep mode.
+        // Can't do this with the nice wrappers on the CommonComponent of the
+        // system device, I suppose since people aren't really connecting to
+        // that device.
+        vrpnConn->register_handler(
+            vrpnConn->register_message_type(vrpn_got_first_connection),
+            &ServerImpl::m_exitIdle, this);
+        vrpnConn->register_handler(
+            vrpnConn->register_message_type(vrpn_dropped_last_connection),
+            &ServerImpl::m_enterIdle, this);
     }
 
-    ServerImpl::~ServerImpl() { stop(); }
+    ServerImpl::~ServerImpl() {
+#if 0
```

Comment:
Typically handlers are unregistered - don't have easy access to the pointer or other things to unregister here,but it didn't seem to cause a problem. The commenting out was my abdication of a decision, probably a poor choice. I can stash a functor to unregister.